trio/_subprocess/unix_pipes.py

```diff
@@ -0,0 +1,119 @@
+import os
+from typing import Tuple
+
+from .. import _core, BrokenStreamError
+from .._abc import SendStream, ReceiveStream
+
+__all__ = ["PipeSendStream", "PipeReceiveStream", "make_pipe"]
+
+
+class _PipeMixin:
+    def __init__(self, pipefd: int, *, set_non_blocking: bool = True):
+        if not isinstance(pipefd, int):
+            raise TypeError(
+                "{0.__class__.__name__} needs a pipe fd".format(self)
+            )
+
+        self._pipe = pipefd
+        self._closed = False
+
+        if set_non_blocking:
```

Comment:
I don't think there's any reason for this to be configurable? We can just do it unconditionally.